cfme/intelligence/reports/reports.py

```diff
@@ -267,6 +270,30 @@ def is_displayed(self):
         )
 
 
+class ImportExportCustomReportsView(CloudIntelReportsView):
+    title = Text("#explorer_title_text")
+    subtitle = Text(locator=".//div[@id='main_div']/h2")
+
+    upload_file = FileInput(id="upload_file")
+    upload_button = InputButton("commit")
+
+    overwrite = Checkbox("overwrite")
+    preserve_owner = Checkbox("preserve_owner")
+
+    items_for_export = Select(id="choices_chosen")
+    export_button = Button(id="export_button")
+
+    @property
+    def is_displayed(self):
+        return (
```

Comment:
Often it happens that something in CFME changes and the `is_displayed` doesn't return `True`.  Then one needs to find why. If you created
```suggestion
result = (
            self.in_intel_reports,
            self.title.text == "Import / Export",
            self.subtitle.text == "Custom Reports",
            self.import_export.tree.currently_selected == ["Import / Export", "Custom Reports"])
)
return all(results)
```
Then it would be slighly easier to debug why the return value is not what is expected. In the debuger we could just check the results tuple

Reply:
Hmm, that's nice! Will add that. Thanks @jaryn  :)

Reply:
To be clear, there is also a clear downside to the above strategy. While useful for debugging, it has an impact on efficiency and time cost of the UI checks.

When chaining together `and` statements in python, once one statement returns `False`, the chain stops. So if `in_intel_reports` is `False`, and we store and then call `all()`, we still run every other check, reading multiple widgets and making string/list comparisons.  If we instead are chaining those calls with `and`, we would have saved the additional time of reading 3 other widgets and doing 3 str/list comparisons.

This is why we use chained `and` pattern in **every** `is_displayed` method in CFME, for efficiency. If you have a failing check and need to use the debugger, then you can easily make this change if that helps you debug, to determine what's failing.  You can also just set a breakpoint and read all the widgets to compare them manually/visually.